invenio_communities/communities/services/components.py

```diff
@@ -169,3 +172,78 @@ def create(self, identity, data=None, record=None, **kwargs):
 
         # Invalidate the membership cache
         on_membership_change(identity=identity)
+
+
+class OAISetComponent(ServiceComponent):
+    """Service component for OAI set integration."""
+
+    _oai_sets_prefix = "community"
+
+    def _retrieve_set(self, record, **kwargs):
+        return OAISet.query.filter(OAISet.spec == self._create_set_spec(record.get("id"))).first()
```

Comment:
```suggestion
        return OAISet.query.filter(OAISet.spec == self._create_set_spec(record.pid.pid_value)).first()
```

Reply:
The function now only has an `id` parameter and passing the right value depends on the context. For renaming, the old `id` has to be used so it can be deleted before the new `id` is used for creation.